offline_cam/plugin.cpp

```diff
@@ -38,118 +33,51 @@ class offline_cam : public threadloop {
 
     virtual void _p_one_iteration() override {
         duration time_since_start = _m_rtc->now().time_since_epoch();
-        duration begin            = time_since_start;
-        ullong   lookup_time      = std::chrono::nanoseconds{time_since_start}.count() + dataset_first_time;
+        // duration begin            = time_since_start;
+        ullong lookup_time = std::chrono::nanoseconds{time_since_start}.count() + dataset_first_time;
         std::map<ullong, sensor_types>::const_iterator nearest_row;
 
-#ifndef INFINITE_QUEUE
+        // "std::map::upper_bound" returns an iterator to the first pair whose key is GREATER than the argument.
         auto after_row = _m_sensor_data.upper_bound(lookup_time);
 
-        if (after_row == _m_sensor_data.cend()) { // || !(after_row->second.cam0 && after_row->second.cam1)
-    #ifndef NDEBUG
+        if (after_row == _m_sensor_data.cend()) {
+#ifndef NDEBUG
             std::cerr << "Time " << lookup_time << " (" << _m_rtc->now().time_since_epoch().count() << " + "
```

Comment:
since the goal of this is to check if the current iteration is the last one, maybe we need to adjust the debug message to reflect that accordingly